include/optimization/cutting_plane.h

```diff
@@ -0,0 +1,1049 @@
+            polytope.put_mat_coeff(i, j, c(j));
+        }
+
+        //add  < c,  point >  in last row of b
+        NT _b = c.dot(point.getCoefficients());
+        polytope.put_vec_coeff(polytope.num_of_hyperplanes() - 1, _b);
+    }
+
+
+    /**
+     * Store in dotProducts the dot product of c with all points in points
+     *
+     * @tparam Point class Point
+     * @tparam NT The numeric type
+     * @param c a vector holding the coefficients of the object function
+     * @param points a list of points
+     * @param dotProducts a list to store the dot products
+     */
+    template<class Point, typename NT>
+    void dotProducts(VT &c, std::list<Point> &points, std::list<NT> &dotProducts) {
+        for (auto p : points)
+            dotProducts.push_back(c.dot(p.getCoefficients()));
+    }
+
+
+    /**
+     * Find the point that minimizes the object function out of a list of points
+     *
+     * @tparam Point class Point
+     * @tparam NT The numeric type
+     * @param c a vector holding the coefficients of the object function
+     * @param randPoints a list of points
+     * @return point wich minimizes the object function
+     */
+    template<class Point, typename NT>
+    Point getMinimizingPoint(VT &c, std::list<Point> &randPoints) {
+        typename std::list<Point>::iterator it = randPoints.begin();
+
+        NT temp, min;
+        Point minPoint = *it;
+
+        min = c.dot(it->getCoefficients());
+        it++;
+
+        for (; it != randPoints.end(); it++) {
+            temp = c.dot(it->getCoefficients());
+
+            if (temp < min) {
+                min = temp;
+                minPoint = *it;
+            }
+        }
+
+        return minPoint;
+    }
+
+
+    /**
+     * Find the two point that minimize the object function the most out of a list of points
+     *
+     * @tparam Point class Point
+     * @tparam NT The numeric type
+     * @param c a vector holding the coefficients of the object function
+     * @param randPoints a list of points
+     * @return a pair of points which minimize the object function (first point minimizes the most)
+     */
+    template<class Point, typename NT>
+    std::pair<Point, Point> getPairMinimizingPoint(VT &c, std::list<Point> &randPoints) {
+        class std::list<Point>::iterator points_it = randPoints.begin();
+        class std::list<Point>::iterator minPoint, minPoint2;
+
+        NT temp, min, min2;
+        minPoint = points_it;
+
+        min = c.dot(points_it->getCoefficients());
+
+        points_it++;
+
+        min2 = c.dot(points_it->getCoefficients());
+        minPoint2 = points_it;
+
+        if (min2 < min) {
+            temp = min2;
+            min2 = min;
+            min = temp;
+            typename std::list<Point>::iterator tempPoint = minPoint;
+            minPoint = minPoint2;
+            minPoint2 = tempPoint;
+        }
+
+        points_it++;
+
+        for (; points_it != randPoints.end(); points_it++) {
```

Comment:
please be consistent and use the same name for iterators as in the function above